### PR TITLE
fix: Use configuration values to ignorePatterns.

### LINKED ISSUE #3872: Use configuration values to the patterns that are always ignored (`ignorePatterns`)
**Is your feature request related to a problem? Please describe.**
if you change a file use by stryker (eg. `tempDirName`), you must remove old file and add new file in `ignorePatterns`:

```JSON
{
    "tempDirName": ".tmp/stryker",
    "ignorePatterns": ["!.stryker-tmp", ".tmp/stryker"]
}
```

**Describe the solution you'd like**
Use configuration values to the patterns that are always ignored (`ignorePatterns`):
`['node_modules', '.git', incrementalFiles, htmlReporter.fileName, jsonReporter.fileName, '*.tsbuildinfo', '/stryker.log', tempDirName]`

You can use this configuration:

```JSON
{
    "tempDirName": ".tmp/stryker"
}
```

### PR DESCRIPTION
**Pull request by regseb**
Fix #3872

- Replace `"/reports"` by `incrementalFiles` in `ignorePatterns` (`tempDirName` was already in use).
- Describe `tempDirName` and `incrementalFiles` in documentation.

I didn't add `htmlReporter.fileName` and `jsonReporter.fileName` because I don't have access to these variables in the constructor.

### CODE REVIEW

File: packages/core/src/fs/project-reader.ts [hunk 1]
```diff
@@ -43,7 +43,7 @@ export class ProjectReader {
     { mutate, tempDirName, ignorePatterns, incremental, incrementalFile, force }: StrykerOptions
```
Comment (nicojs) [inline packages/core/src/fs/project-reader.ts:43]:
```suggestion
    { mutate, tempDirName, ignorePatterns, incremental, incrementalFile, force, htmlReporter, jsonReporter }: StrykerOptions
```

You can pick off the `htmlReporter` and `jsonReporter` options from the `StrykerOptions` here.

File: packages/core/src/fs/project-reader.ts [hunk 2]
```diff
@@ -43,7 +43,7 @@ export class ProjectReader {
...
   ) {
     this.mutatePatterns = mutate;
-    this.ignoreRules = [...ALWAYS_IGNORE, tempDirName, ...ignorePatterns];
+    this.ignoreRules = [...ALWAYS_IGNORE, tempDirName, incrementalFile, ...ignorePatterns];
```
Comment (nicojs) [inline packages/core/src/fs/project-reader.ts:46]: ... And add them here.

**Review by nicojs — CHANGES_REQUESTED** [review]
Thanks for picking this one up, and sorry it took me so long to respond. You should be able to add the output files from the HTML and JSON reporter quite easily; see my remark.

File: packages/core/src/fs/project-reader.ts (hunk 2 above)
Comment (regseb) [reply packages/core/src/fs/project-reader.ts]: I added `htmlReporter.fileName` and `jsonReporter.fileName`.

**Review by nicojs — APPROVED** [review]
This looks great! I love the additional test and changes there.

**Pull request merged**